.circleci/config.yml

```diff
@@ -75,14 +75,27 @@ commands:
           destination: test_results
 
 jobs:
-  unit_test:
+  linting_checks:
     <<: *job_defaults
     steps:
       - test_setup
       - run:
-          name: Check Python lint
+          name: Check Python lint with yapf
           working_directory: ~/curation
           command: tools/yapf_lint.sh
+      - run:
+          name: Pylint on data_steward
+          working _directory: ~/curation
+          command: pylint -E data_steward/
+      - run:
+          name: Pylint on tests
+          working_directory: ~/curation
+          command: pylint -E tests/
+      - test_teardown
```

Comment:
Most of the steps in `test_teardown` (handling test and coverage results) are not relevant for linting and will only add noise. I think `save_cache` is still helpful though and we can refactor such that we just do that last.

Reply:
updated